design/one-pager-resource-status-improvements.md

```diff
@@ -0,0 +1,91 @@
+
+    Start --> NeedsDelete
+    NeedsDelete -->|Yes| Delete
+    NeedsDelete -->|No| NeedsCreate
+    NeedsCreate -->|Yes| Create
+    NeedsCreate -->|No| UpToDate
+    UpToDate -->|Yes| Observe
+    UpToDate -->|No| Update
+```
+
+#### Solution
+Implement an `Updating` status, mimicking the `Creating` status. Setting this status every time an update request either succeeds or fails, informing a user that an update request has been requested, but we have not verified if the resource is compliant with the updated desired state.
+
+This also requires updating the reconciler to immediately requeue a reconcile after a successfuly requested update. This ensures that an updated resource is immediatly checked for compliance, to give user quick feedback on if the resource is `Available` again after the update.
+
+Note that this would technically change the definition of the `Ready` condition. However, this definition is not very well documented, so I am not sure how many know about it.  
+Old definition:
+>* Ready - The external resource is ready to use, (where applicable) according to its API.  
+
+New definiton:
+>* Ready - The external reosurce is compliant with the desired state and is ready to use. 
+
+##### Alternative solution
+I recommend the original solution with an Updating status for consistency and ease of use. However, as a possible alternative, we could forgo the Updating status entirely and simply trigger an immediate reconcile after successful updates.  This would retain the current Ready definition and allow providers complete control over how they implement a similar status if desired.
+
+#### Benefits
+* **Immediate Compliance Feedback**: All users will quickly know whether their resource matches the desired state, providing greater clarity and confidence.
+* **Streamlined Understanding**: Eliminates the need to interpret multiple fields to assess resource status.
+* **Focus on User-Centric Status**: While the external API's availability remains important, these changes emphasize a focus on the user's primary concern – whether their resource configuration is correct.
+
+### 1: Improving Providers Control Over Condition Messages
+#### Problem
+Crossplane currently has primary control over the `Ready` condition, limiting the ability of providers to communicate detailed, human-readable status information to users via the `Message` field. While Crossplane's consistent conditions create predictable expectations for users, this generic nature leaves valuable context underutilized. Important status information for users and developers is often relegated to Kubernetes events and logs, which can be less convenient to access and interpret.
+
+#### Solution
+Modify the existing condition helper functions in Crossplane to optionally accept a string argument intended for the `Message` field of a condition. This would empower provider authors to provide human-readable summaries when setting condition states.
```

Comment:
I understand, the first sentence in the solution wasn't optimal 😄 
The limitation stems from the crossplane-runtime setting the `Ready` condition most of the time, without a message. 

Say something went wrong when creating an external resource. I would then want to create something like, to give good feedback to my users in a human readable format.
```
Type: Ready
Status: False
Reason: Creating
Message: This and that went wrong, etc.
```
If I create this condition and set it on my MR, it would immediately be overwritten by crossplane-runtime with a similar condition, without the message. 

So my intent with this was basically to respect the message field, but it could have been explained better!